test/unit-test/FreeRTOS_IPv6_Utils/FreeRTOS_IPv6_Utils_utest.c

```diff
@@ -0,0 +1,367 @@
+    xICMPHeaderIPv6.ucTypeOfMessage = ipICMP_Incorrect_Msg_IPv6;
+    xProtocolHeaders.xICMPHeaderIPv6 = xICMPHeaderIPv6;
+    xSet.uxIPHeaderLength = ipSIZE_OF_IPv6_HEADER;
+    xSet.pxProtocolHeaders = &xProtocolHeaders;
+
+    usReturn = prvChecksumICMPv6Checks( uxBufferLength, &xSet );
+
+    TEST_ASSERT_EQUAL( ipFail_INVALID_LENGTH, usReturn );
+    TEST_ASSERT_EQUAL( ipINVALID_LENGTH, xSet.usChecksum );
+}
+
+/**
+ * @brief This function verify sending an valid length.
+ */
+void test_prvChecksumICMPv6Checks_Default_ValidLength( void )
+{
+    uint16_t usReturn;
+    size_t uxBufferLength = ipSIZE_OF_ETH_HEADER + ipSIZE_OF_IPv6_HEADER + ipSIZE_OF_ICMPv6_HEADER + 1;
+    struct xPacketSummary xSet;
+    ICMPHeader_IPv6_t xICMPHeaderIPv6;
+    ProtocolHeaders_t xProtocolHeaders;
+
+    memset( &xSet, 0, sizeof( struct xPacketSummary ) );
+    xICMPHeaderIPv6.ucTypeOfMessage = 0;
+    xProtocolHeaders.xICMPHeaderIPv6 = xICMPHeaderIPv6;
+    xSet.uxIPHeaderLength = ipSIZE_OF_IPv6_HEADER;
+    xSet.pxProtocolHeaders = &xProtocolHeaders;
+
+    usReturn = prvChecksumICMPv6Checks( uxBufferLength, &xSet );
+
+    TEST_ASSERT_EQUAL( usReturn, 0 );
+    TEST_ASSERT_EQUAL( ipSIZE_OF_ICMPv6_HEADER, xSet.uxProtocolHeaderLength );
+}
+
+/**
+ * @brief This function verify sending an invalid length
+ *        for ipICMP_PING_REQUEST_IPv6.
+ */
+void test_prvChecksumICMPv6Checks_PingReq_InvalidLength( void )
+{
+    uint16_t usReturn;
+    size_t uxBufferLength = ipSIZE_OF_ETH_HEADER + ipSIZE_OF_IPv6_HEADER + sizeof( ICMPEcho_IPv6_t ) - 1;
+    struct xPacketSummary xSet;
+    ICMPHeader_IPv6_t xICMPHeaderIPv6;
+    ProtocolHeaders_t xProtocolHeaders;
+
+    memset( &xSet, 0, sizeof( struct xPacketSummary ) );
+    xProtocolHeaders.xICMPHeaderIPv6 = xICMPHeaderIPv6;
+    xICMPHeaderIPv6.ucTypeOfMessage = ipICMP_PING_REQUEST_IPv6;
+    xSet.uxIPHeaderLength = ipSIZE_OF_IPv6_HEADER;
+    xSet.pxProtocolHeaders = &xProtocolHeaders;
+
+    usReturn = prvChecksumICMPv6Checks( uxBufferLength, &xSet );
+
+    TEST_ASSERT_EQUAL( ipFail_INVALID_LENGTH, usReturn );
+    TEST_ASSERT_EQUAL( ipINVALID_LENGTH, xSet.usChecksum );
+}
+
+/**
+ * @brief This function verify sending an valid length
+ *        for ipICMP_PING_REQUEST_IPv6.
+ */
+void test_prvChecksumICMPv6Checks_PingReq_ValidLength( void )
+{
+    uint16_t usReturn;
+    size_t uxBufferLength = ipSIZE_OF_ETH_HEADER + ipSIZE_OF_IPv6_HEADER + sizeof( ICMPEcho_IPv6_t ) + 1;
+    struct xPacketSummary xSet;
+    ICMPHeader_IPv6_t xICMPHeaderIPv6;
+    ProtocolHeaders_t xProtocolHeaders;
+
+    memset( &xSet, 0, sizeof( struct xPacketSummary ) );
+    xProtocolHeaders.xICMPHeaderIPv6 = xICMPHeaderIPv6;
+    xICMPHeaderIPv6.ucTypeOfMessage = ipICMP_PING_REQUEST_IPv6;
+    xSet.uxIPHeaderLength = ipSIZE_OF_IPv6_HEADER;
+    xSet.pxProtocolHeaders = &xProtocolHeaders;
+
+    usReturn = prvChecksumICMPv6Checks( uxBufferLength, &xSet );
+
+    TEST_ASSERT_EQUAL( pdFALSE, usReturn );
```

Comment:
similar to ipFail_INVALID_LENGTH, we can have 
ipSUCCESS ( 0 )
ipFAILURE   ( 1 )

Reply:
Sure